src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1700,17 +1701,30 @@ private bool isFieldStatic(CORINFO_FIELD_STRUCT_* fldHnd)
         {
             return HandleToObject(fldHnd).IsStatic;
         }
-
+        private enum ICorDebugInfo
+        {
+            NO_MAPPING = -1, // -- The IL offset corresponds to no source code (such as EH step blocks).
+            PROLOG     = -2, // -- The IL offset indicates a prolog
+            EPILOG     = -3  // -- The IL offset indicates an epilog
+        }
         public void SetSequencePoints(IEnumerable<ILSequencePoint> ilSequencePoints)
         {
             Debug.Assert(ilSequencePoints != null);
             Dictionary<int, SequencePoint> sequencePoints = new Dictionary<int, SequencePoint>();
-
-            foreach (var point in ilSequencePoints)
+            var ilSequencePointsList = ilSequencePoints.ToList();
+            foreach (var point in ilSequencePointsList)
             {
-                sequencePoints.Add(point.Offset, new SequencePoint() { Document = point.Document, LineNumber = point.LineNumber });
-            }
-
+                int offset = point.Offset;
+                if (point.Equals(ilSequencePointsList.First()))
+                {
+                    offset = (int)ICorDebugInfo.PROLOG;
+                }
+                else if (point.Equals(ilSequencePointsList.Last()))
+                {
+                    offset = (int)ICorDebugInfo.EPILOG;
```

Comment:
Fixed, as I studied from SOS\Strike\util.cpp, if there are several epilogs they are pointed to one IL offset.